### PR TITLE
#281 Bulking support

### PR DESCRIPTION
**Pull request by joaofsl**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-pulse/issues/281|
| Decisions | Add missing methods to create a bulk order.|

### CODE REVIEW

**Comment by BeeMargarida** [comment]
@joaofsl Missing changelog, also this needs to be ported to ripe-api in Python

**Comment by BeeMargarida** [comment]
@joaofsl Please link the issue in the changelog entry

File: src/js/api/bulk-order.js [hunk 1]
```diff
@@ -416,6 +416,59 @@ ripe.Ripe.prototype.cancelBulkOrderP = function(number, options) {
...
+/**
+ * Creates a bulk order on RIPE Core.
+ *
+ * @param {String} name The name for the bulk order.
+ * @param {String} brand The brand of the bulk order.
+ * @param {Object} orders The list of the order ids for the bulk order.
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} Resulting information for the callback execution.
+ */
+ripe.Ripe.prototype.importBulkOrder = function(name, brand, orders, options, callback) {
```
Comment (BeeMargarida) [inline src/js/api/bulk-order.js:447]: the name brand and orders could not be inside the options, like `importOrder`?

File: src/js/api/bulk-order.js [hunk 2]
```diff
@@ -416,6 +416,59 @@ ripe.Ripe.prototype.cancelBulkOrderP = function(number, options) {
...
+};
+
+/**
+ * Creates a bulk order on RIPE Core.
+ *
+ * @param {String} name The name for the bulk order.
+ * @param {String} brand The brand of the bulk order.
+ * @param {Object} orders The list of the order ids for the bulk order.
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} Resulting information for the callback execution.
```
Comment (BeeMargarida) [inline src/js/api/bulk-order.js:462]:
```suggestion
 *  @returns {Promise} The bulk order's data.

```

File: src/js/api/bulk-order.js [hunk 3]
```diff
@@ -416,6 +416,59 @@ ripe.Ripe.prototype.cancelBulkOrderP = function(number, options) {
...
+    });
+};
+
+/**
+ * Creates a bulk order on RIPE Core.
+ *
+ * @param {String} name The name for the bulk order.
+ * @param {String} brand The brand of the bulk order.
+ * @param {Object} orders The list of the order ids for the bulk order.
+ * @param {Function} callback Function with the result of the request.
```
Comment (BeeMargarida) [inline src/js/api/bulk-order.js:444]: does not receive callback

File: src/js/api/bulk-order.js [hunk 4]
```diff
@@ -416,6 +416,59 @@ ripe.Ripe.prototype.cancelBulkOrderP = function(number, options) {
...
+        auth: true
+    });
+};
+
+/**
+ * Creates a bulk order on RIPE Core.
+ *
+ * @param {String} name The name for the bulk order.
+ * @param {String} brand The brand of the bulk order.
+ * @param {Object} orders The list of the order ids for the bulk order.
```
Comment (BeeMargarida) [inline src/js/api/bulk-order.js:443]:
```suggestion
 * @param {Object} orders The list of the order IDs for the bulk order.

```

**Review by BeeMargarida — CHANGES_REQUESTED** [review]
Small thing

**Review by BeeMargarida — APPROVED** [review]

**Review by hugo-gomes — COMMENTED** [review]
@joaofsl any update on this?

File: src/js/api/bulk-order.js [hunk 5]
```diff
@@ -416,6 +416,63 @@ ripe.Ripe.prototype.cancelBulkOrderP = function(number, options) {
     });
 };
 
+/**
+ * @ignore
+ * @see {link https://docs.platforme.com/#order-endpoints-import}
```
[the remark this answers is not in the document]
Comment (joaofsl) [reply src/js/api/bulk-order.js:421]: Yes, we don't have any documentation regarding the bulk orders

[the remark this answers is not in the document]
Comment (BeeMargarida) [reply src/js/api/bulk-order.js:421]: @hugo-gomes A release is needed in the docs, the bulk order documentation is already done but not yet deployed

[the remark this answers is not in the document]
Comment (hugo-gomes) [reply src/js/api/bulk-order.js:421]: @ripe-tech/tech-devops can we "release the kraken" in the near future? :)

**Review by joamag — APPROVED** [review]

**Pull request merged**